conans/paths/package_layouts/package_cache_layout.py

```diff
@@ -110,6 +110,43 @@ def package(self, pref):
         assert pref.ref == self._ref, "{!r} != {!r}".format(pref.ref, self._ref)
         return os.path.join(self._base_folder, PACKAGES_FOLDER, pref.id)
 
+    @contextmanager
+    def set_dirty_context_manager(self, pref):
+        pkg_folder = os.path.join(self._base_folder, PACKAGES_FOLDER, pref.id)
+        set_dirty(pkg_folder)
+        yield
+        clean_dirty(pkg_folder)
+
+    def package_is_dirty(self, pref):
+        pkg_folder = os.path.join(self._base_folder, PACKAGES_FOLDER, pref.id)
+        return is_dirty(pkg_folder)
+
+    def package_set_dirty(self, pref):
+        pkg_folder = os.path.join(self._base_folder, PACKAGES_FOLDER, pref.id)
+        return set_dirty(pkg_folder)
+
+    def package_clean_dirty(self, pref):
+        pkg_folder = os.path.join(self._base_folder, PACKAGES_FOLDER, pref.id)
+        return clean_dirty(pkg_folder)
```

Comment:
If `package_remove` removes also the `.dirty` sentinel for the package-ID, I think that methods `package_set_dirty` and `package_clean_dirty` won't be needed, only the context manager one, which is much safer for this kind of job.

Reply:
Yes, I have considered doing this. But this might slightly change the behavior, so I wanted to wait for feedback first.
Should I try in this PR or maybe in a new one after this?